app/views.py

```diff
@@ -0,0 +1,160 @@
+from django.core.urlresolvers import reverse
+# from django.shortcuts import render
+from django.contrib import messages
+from django.contrib.auth.models import User
+
+
+@csrf_protect
+def register(request):
+    if request.method == 'POST':
+        form = RegistrationForm(request.POST)
+        if form.is_valid():
+            try:
+                date = datetime.datetime(
+                    year=int(form.cleaned_data['year']),
+                    month=int(form.cleaned_data['month']),
+                    day=int(form.cleaned_data['day'])
+                )
+                dob = date.strftime('%Y-%m-%d')
+                user = User.objects.create_user(
+                    username=form.cleaned_data['username'],
+                    password=form.cleaned_data['password1'],
+                )
+                profile = UserProfile(user=user, date_of_birth=dob)
+                profile.save()
+                return HttpResponseRedirect('/')
+            except ValueError:
+                messages.error(request, 'The day is out of range for month.')
+        variables = RequestContext(request, {
+            'form': form
+        })
+        return render_to_response(
+            'registration/register.html',
+            variables,
+        )
+    form = RegistrationForm()
+    variables = RequestContext(request, {
+        'form': form
+    })
+    return render_to_response(
+        'registration/register.html',
+        variables,
+    )
+
+
+def register_success(request):
+    return HttpResponseRedirect('/')
+
+
+def logout_page(request):
+    logout(request)
+    return HttpResponseRedirect('/')
+
+
+@login_required
+def home(request):
+    return HttpResponseRedirect('/')
+
+
+class MyProfileView(TemplateView):
+    """
+    Enables viewing of the user's profile in the HTML site, by the profile
+    owner.
+    """
+    template_name = 'app/viewprofile.html'
+
+    def get_context_data(self, **kwargs):
+        """ Retrieve the user profile
+        """
+        context = super(MyProfileView, self).get_context_data(**kwargs)
+        user = self.request.user
+        profile = user.profile
+        context['username'] = user.username
+
+        if profile.alias:
+            context['alias'] = profile.alias
+        else:
+            context['alias'] = 'Anonymous'
+
+        context['date_of_birth'] = profile.date_of_birth
+        return context
+
+
+class MyProfileEdit(FormView):
+    """
+    Enables editing of the user's profile in the HTML site
+    """
+    form_class = EditProfileForm
+    template_name = 'app/editprofile.html'
+
+    def get_initial(self):
+        initial = self.initial.copy()
+        user = self.request.user
+        profile = user.profile
+        initial['username'] = user.username
+
+        if profile.alias:
+            initial['alias'] = profile.alias
+        else:
+            initial['alias'] = 'Anonymous'
+
+        initial['date_of_birth'] = profile.date_of_birth
+        return initial
```

Comment:
See my comment above about a context processor that adds a `profile` to the default template context.